openhtf/__init__.py

```diff
@@ -84,32 +84,56 @@ def __call__(self, test_record):  # pylint: disable=invalid-name
       f.write(self.encode(as_dict))
 
 
-def TestPhase(timeout_s=None, run_if=None):  # pylint: disable=invalid-name
-  """Decorator to wrap a test phase function with the given options.
-
-  Args:
+class TestPhaseOptions(mutablerecords.Record(
+    'TestPhaseOptions', [], {'timeout_s': None, 'run_if': None})):
+  """
+  Attributes:
     timeout_s: Timeout to use for the phase, in seconds.
     run_if: Callback that decides whether to run the phase or not.  The
       callback will be passed the phase_data the phase would be run with.
 
-  Returns:
-    A wrapper function that takes a phase_func and returns a
-        TestPhaseInfo for it with the given options set.
+  Example Usage:
+    @TestPhaseOptions(timeout_s=1)
+    def PhaseFunc(test):
+      pass
   """
 
-  def Wrap(phase_func):  # pylint: disable=invalid-name
-    """Attach the given options to the phase_func."""
+  def __call__(self, phase_func):
+    phase = TestPhaseInfo.WrapOrReturn(phase_func)
+    phase.options = self
+    return phase
 
-    # Test Phases must take at least one argument (the phase data tuple).
-    if len(inspect.getargspec(phase_func).args) < 1:
-      raise InvalidTestPhaseError(phase_func, 'Not enough args')
+PhasePlug = mutablerecords.Record('PhasePlug', ['name', 'cls', 'update_kwargs'])
 
-    if timeout_s is not None:
-      phase_func.timeout_s = timeout_s
-    if run_if is not None:
-      phase_func.run_if = run_if
-    return phase_func
-  return Wrap
+
+class TestPhaseInfo(mutablerecords.Record(
```

Comment:
Also, the only line in CONTRIBUTING.md regarding this is here:

    We welcome patches and contribution to OpenHTF. It's as easy as submitting a pull request with your changes in order to kick off the review process. But first, please read through the rest of this doc.

So I followed the one and only request available.